src/components/charts-generic/histogram/median.tsx

```diff
@@ -13,21 +14,32 @@ export const HistogramMedian = ({ label }: { label: string }) => {
     yScale,
     xAxisLabel,
     groupedBy,
+    binMeta,
   } = useChartState() as HistogramState;
   const { margins } = bounds;
   const { labelColor, domainColor, labelFontSize, fontFamily } =
     useChartTheme();
   const formatCurrency = useFormatCurrency();
 
   let medianX: number | undefined = undefined;
-  if (m !== undefined) {
-    medianX = xScale(m);
-    if (groupedBy) {
-      const binStart = Math.floor(m / groupedBy) * groupedBy;
-      const binEnd = binStart + groupedBy;
-      const binCenter = (binStart + binEnd) / 2;
-      medianX = xScale(binCenter);
+
+  if (m !== undefined && groupedBy && binMeta) {
+    // Find the bin containing the median
+    const bin = binMeta.find((b) => !b.isNoData && m >= b.x0 && m <= b.x1);
+    if (bin) {
+      // Use band scale to get the center of the bar
+      const bandDomain = binMeta.map((b, i) => b.label ?? String(i));
+      const bandScale = scaleBand<string>()
+        .domain(bandDomain)
+        .range([0, bounds.chartWidth]);
```

Comment:
Seems like the bandScale should be returned as part of the useChartState. I would guess there is already a scaleBand there ?